### ISSUE TITLE
'Missing' ATLAS targets, discovered by other surveys

### ISSUE THREAD

**Issue by katemaguire** [genghisken/ps1#81]
I’ve been looking through the objects in the ePESSTO inbox and classification queue that are not listed as being detected by ATLAS but are registered by other surveys. This was prompted by AT2018jq that was in the ATLAS garbage due to a low realbogus score but has a nice light curve (it now registered).

Turns out a lot of other objects are in the ATLAS database e.g. in the classification queue

AT2018qq - discovered by ASASSN, in the stars list for ATLAS at 2.92” offset with discoveries over the last few nights. No realbogus.
AT2018qp - discovered by ASASSN, in ATLAS at 2.52” offset with multiple photometry points over the last two weeks. Don’t know what list it is in? No realbogus.
AT2018av - discovered by PS, in ATLAS as two separate targets at 0.19" and 1.05” offsets with photometry coeval with PS discovery showing clear light curve. In the garbage for both. This one is tricky since it’s an NT.
AT2018fj - discovered by PS, in ATLAS garbage list at 0.6” offset. Nice light curve over last few weeks. Realbogus 0.08

I had spoken with Ken about missing some in the garbage because of their low realbogus objects and may be be improved by his updated training set. However, it does seem like there may be a couple of other issues such as for AT2018qp - where is it or what happened to it?

Ken also said he would update the code so that objects can be removed from the garbage more easily. It would be very good to even be able to easily go through the ePESSTO classification list and check for ATLAS objects at the same location and then register them. It would definitely help prioritise the classifications because some show that they have been visible for a while and therefore, old.

**Comment by thespacedoctor** [comment]
## AT2018qq

I'll comment on this source as this is a Sherlock issue to some extent. You can find AT2018qq in the marshall [here](http://www.pessto.org/marshall/transients/search?q=AT2018qq). The story of the source within the ATLAS database is a little complicated, but I think this is what happened:

1. The source got added to the ATLAS database and made it to the eyeball list when it passed the threshold for the number of detections.
2. Sherlock classified it with the average coordinates it had at that stage:

```bash
> sherlock -N match -- 149.63674    -27.83077 ~/.config/sherlock/sherlock_atlas4_all.yaml
Transient's Predicted Classification: VS

Suggested Associations:
+-------------------+-------+------------+-----------------------+-------------------------------+------------------------+---------------------------+-------------+--------------+-------------------+--------------------------+------------------+-----------+-------+--------+------------+---------+-----------------------------+
| association type  | rank  | rankScore  | catalogue table name  | catalogue object id           | catalogue object type  | catalogue object subtype  | raDeg       | decDeg       | separationArcsec  | physical separation kpc  | direct distance  | distance  | z     | Mag    | MagFilter  | MagErr  | classification reliability  |
+-------------------+-------+------------+-----------------------+-------------------------------+------------------------+---------------------------+-------------+--------------+-------------------+--------------------------+------------------+-----------+-------+--------+------------+---------+-----------------------------+
| VS                | 1     | 2000.84    | Gaia DR1              | 5465029057308325248           | star                   |                           | 149.636743  | -27.8308648  | 0.34              |                          |                  |           |       | 19.56  | G          |         | association                 |
| SN                | 2     | 3000.43    | NED                   | GALEXASC J095832.82-274950.4  | galaxy                 | UvS                       | 149.636792  | -27.83069    | 0.33              |                          |                  |           |       |        |            |         | annotation                  |
| VS                | 3     | 3000.84    | Gaia DR1              | 5465029057308325248           | star                   |                           | 149.636743  | -27.8308648  | 0.34              |                          |                  |           |       | 19.56  | G          |         | annotation                  |
| UNCLEAR           | 4     | 3000.94    | GSC v2.3              | S68F002945                    | unknown                | 3                         | 149.636791  | -27.830819   | 0.24              |                          |                  |           |       |        |            |         | annotation                  |
| UNCLEAR           | 5     | 3010.83    | NED                   | 1RXS J095833.6-274953         | other                  | XrayS                     | 149.64      | -27.83153    | 10.73             |                          |                  |           |       |        |            |         | annotation                  |
+-------------------+-------+------------+-----------------------+-------------------------------+------------------------+---------------------------+-------------+--------------+-------------------+--------------------------+------------------+-----------+-------+--------+------------+---------+-----------------------------+
```

3. Ken's code saw the VS classification and automatically moved it to the stars list.
4. More detections rolled in and improved the accuracy of the sky-position measurement of the transient. The transient's measured position has moved 2.9 arcsec from its first detected location to the current average location.

If I now run Sherlock on the current average location a more accurate classification is given:

```bash
> sherlock -N match -- 149.63763 -27.83058 ~/.config/sherlock/sherlock_atlas4_all.yaml
Transient's Predicted Classification: SN

Suggested Associations:
+-------------------+-------+------------+-----------------------+-------------------------------+------------------------+---------------------------+-------------+------------+-------------------+--------------------------+------------------+-----------+-------+-----------------------------+
| association type  | rank  | rankScore  | catalogue table name  | catalogue object id           | catalogue object type  | catalogue object subtype  | raDeg       | decDeg     | separationArcsec  | physical separation kpc  | direct distance  | distance  | z     | classification reliability  |
+-------------------+-------+------------+-----------------------+-------------------------------+------------------------+---------------------------+-------------+------------+-------------------+--------------------------+------------------+-----------+-------+-----------------------------+
| SN                | 1     | 3002.80    | NED                   | GALEXASC J095832.82-274950.4  | galaxy                 | UvS                       | 149.636792  | -27.83069  | 2.70              |                          |                  |           |       | annotation                  |
| UNCLEAR           | 2     | 3008.38    | NED                   | 1RXS J095833.6-274953         | other                  | XrayS                     | 149.64      | -27.83153  | 8.28              |                          |                  |           |       | annotation                  |
+-------------------+-------+------------+-----------------------+-------------------------------+------------------------+---------------------------+-------------+------------+-------------------+--------------------------+------------------+-----------+-------+-----------------------------+
```

My suggested solution to this issue would be to:

1. Wait until a transient in the stars list has N-detections but less than M-detections (N=5 and M=10?)
2. Remove the sherlock classification and move the transients back into the eyeball list
3. It the new average position gives a classification other than VS/BS then it will stay in the eyeball list for humans to see

There may be a little more book-keeping needed than this, but I think it's a simple enough solution.

@smarttgit

**Comment by katemaguire** [comment]
Here is another ASASSN (http://www.pessto.org/marshall/transients/search?q=2018tt) one that is in ATLAS in the garbage at realbogus of 0.01: https://star.pst.qub.ac.uk/sne/atlas4/candidate/1095423200143913900/

Clearly rising just offset from spec z=0.06 galaxy

Offset from ASASSN position is large at 3.84".

**Comment by katemaguire** [comment]
AT2018tr (http://www.pessto.org/marshall/transients/search?q=2018tr) as https://star.pst.qub.ac.uk/sne/atlas4/candidate/1105150081250242300/ and https://star.pst.qub.ac.uk/sne/atlas4/candidate/1105150211250238600/ Real bogus of 0.01

Rising offset from spec z=0.045 galaxy.

**Comment by genghisken** [comment]
The web code will normally render the list from which the object came by default if the object is moved to another list (e.g. from garbage to eyeball).  The code is now altered to render the eyeball list by default even if the object has come from the garbage.  This should make things easier to rescue in future.

**Comment by genghisken** [comment]
AT2018qp was originally unpromoted when Kate flagged this object.  Since then it got promoted - and the ditched by a human (on 20th Feb) - almost certainly because some of the diffs did not look convincing.  Rescued and promoted.

**Comment by genghisken** [comment]
AT2018av was accidentally ditched by a human on 9th December.  No idea why.  Subtractions look OK.

**Comment by katemaguire** [comment]
Some more. The following objects were discovered by ASASSN/CRTS and are in ATLAS but not registered:
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140901350303327800/ - clearly rising, not flagged because it doesn't have enough high pvr values maybe?
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1110034071095713200/ - faint but rising
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1050319600105818900/ - six detections but on same night
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1122558591042828200/ - only 1 detection

**Comment by genghisken** [comment]
AT2018fj was ditched my ML because realbogus factors very low.  In this case the object was faint for ATLAS, hence RB factors would have been low. Rescued and promoted.  Difficult one - this will always happen for fainter objects, especially objects which never get particularly bright.

**Comment by smarttgit** [comment]
We're not looking at this in the right way. 

"Rescuing" and "promoting" these objects is not a solution. They are all real and they do have +ve flux in ATLAS. We need to fix why they are not being promoted. 

Correct to first identify why. But we need to collect the reasons together and fix.

**Comment by genghisken** [comment]
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140901350303327800/ - dominated by psc. So, yes, pvr not high enough.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1110034071095713200/ - double issue of being both faint, and having an odd shaped PSF - hence machine learning scored low.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1050319600105818900/ - dominated again by value of psc - hence pvr not high enough to flag.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1122558591042828200/ - we need more detections to flag the object - and even the one detection dominated again by psc.  So would not count to total of flagged detections.  (Maybe we can lower or scale the pvr/ptr threshold if the other dominant parameter is psc.)

**Comment by katemaguire** [comment]
So definitely to catch the likes of AT2018yh (https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140901350303327800/) we need to adjust the psc/pvr/ptf thresholds and/or figure out why psc is so high. This is in a specz=0.023 galaxy so would be included in the planned 'fast-track' list but we shouldn't be missing ones like this in general.

**Comment by katemaguire** [comment]
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1060058990175930400/ aka AT2018xv
I would say this one was moved to garbage because of bad subtractions in the field on the first night but had a quite high real-bogus factor. Can't see image stamps for the subsequent night but might have been better and easier to identify as a real rising transient in those.

**Comment by genghisken** [comment]
This one was a mistake by the eyeballer.  It should not have been ditched.  It's clearly a transient.  I need to educate the eyeballers on the difference between a bad subtraction and a subtraction that contains bright star artifacts.  In this case the bright stars left artifacts, but the fainter objects subtracted OK.  The situation might also have been confused by the PS1 stamp. It shows a galaxy in the position of the transient, but also a red flare star close to, but just below the transient.

**Comment by smarttgit** [comment]
All eyeballers need to be signed up to this page to see where things go wrong - you don't have time to go round everyone trying to explain. Not sustainable. Everybody must see THIS page, and see examples.  Can you ensure everyone is

**Comment by katemaguire** [comment]
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1102031021031343700/ aka AT2018yr Not enough high pvrs?

**Comment by genghisken** [comment]
Yes.  Also centroiding is pretty bad.

**Comment by katemaguire** [comment]
These are taken from the ePESSTO classification queue, where ATLAS didn't discover them but another survey did. I've ignored for now, objects with just one ATLAS detection. 
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140901350303327800/ aka AT2018yh. Detected in multiple epochs, currently ~16.5 mag. Low pvrs again?
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1114613021202318800/ aka AT2018yn
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140338920335841400/ aka AT2018zz. What do the diamonds in the light curves mean? Has "-o" filter?
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140732020220922600/ aka AT2018zj
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1053231601145208600/ aka AT2018qf
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1080302751273459200/ aka AT2018vp
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1142613410015542600/ aka AT2018aas
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1145604000274604600/ aka AT2018aav This one is a catalogue issue, matched with a Gaia DR1 star but clearly in a galaxy. Sherlock got it right.

**Comment by genghisken** [comment]
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140901350303327800/ aka AT2018yh. Detected in multiple epochs, currently ~16.5 mag. Low pvrs again?
yes

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1114613021202318800/ aka AT2018yn
pvr again

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140338920335841400/ aka AT2018zz. What do the diamonds in the light curves mean? Has "-o" filter?
The diamonds indicate negative flux. So -o means negative o flux.  This sometimes happens as the wallpaper is replaced for that sky region.
In this case the galaxy subtractions have left residuals. We need to figure out a way of excluding the residuals and picking up on the high SNR detections. And these detections are almost completely dominated by the psc value.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140732020220922600/ aka AT2018zj
This object is sitting in the possible list.  It was probably placed here because of the initial single detection on 58111 plus the detections on 58158 and 58159.  User wasn't sure it was real at the time, and the cyan detections arrived later. We need to constantly poll the possible list, especially as more detections come in. Don't know why is was regarded as possible - looks OK to me.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1053231601145208600/ aka AT2018qf
Ditched by machine learning - possibly because of elongated PSF in some of the input images.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1080302751273459200/ aka AT2018vp
Ditched by machine learning - down at the detection limit and the machine learning has difficulty with low SNR detections.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1142613410015542600/ aka AT2018aas
The values are evenly split between pvr and ptr. Hence not enough in either to trigger.  Checking for pvr OR ptr would have triggered it.

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1145604000274604600/ aka AT2018aav This one is a catalogue issue, matched with a Gaia DR1 star but clearly in a galaxy. Sherlock got it right.

In this case Sherlock got it wrong in ATLAS, but right in the Marshall.  Coordinates in both places are very similar, so it's probable that Dave has updated the Sherlock algorithm since 14th Feb when the object was triggered in ATLAS.

**Comment by genghisken** [comment]
All of these objects (provided by Michael) are not triggered because of value of psc. Reducing the pvr / ptr trigger threshold to > 199 (i.e. 200) would pass them all.  Note that we need to do a proper analysis of how many more objects will get triggered when we do this with the test data I used for my original ATLAS blog analysis. (https://atlas.fallingstar.com/atlas/trac/wiki/AtlasTransientServer2017 - authentication required.)

https://star.pst.qub.ac.uk/sne/atlas4/candidate/1020459660415207700/
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1090456371144008300/
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1053636560314316700/
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1015609331173829900/
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1195301191411256800/
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1204159081130937200/
https://star.pst.qub.ac.uk/sne/atlas4/candidate/1033145320393651500/

**Comment by katemaguire** [comment]
Discovered by ASASSN, AT2018aqe but nice ATLAS light curve. pvr values below cut off. https://star.pst.qub.ac.uk/sne/atlas4/candidate/1143553671351718000/

**Comment by genghisken** [comment]
OK. I've been doing an ingest into a test database.  I've finished testing HKO data, but I'd like to test MLO data before going live.  Result is that if I reduce the pvr/ptr threshold down to 200 from 500, we get almost double the number of object passing the thresholds.  If this holds for MLO, then we should go with the new cuts. However, this most recent ASASSN object would not pass even the new threshold, since only a handful of individual detections (on separate nights) have values above 200.

In which case I should also look at testing how much more data we get if we reduce the threshold to 100.

**Comment by smarttgit** [comment]
No, don't go to 100.  

Let's look at 200. Don't understand why you say it would not be flagged. Don't use words like "handful" here !! You have a number in your code, let's be quantitative so we can understand 

- It should be flagged after 3 detections with pvr >200 or ptr >200. Am I not correct ? So then why would it not be flagged on  58207.4623442, which is the 3rd quality detection

- I thought we have 3 detection requirement, with no further insistence on same night. Is this not correct? 

And when you say you get double the number of objects - have you already removed all the VS objects. These would likely be a large fraction of high PSC values. 

So in summary, let's move it to 200 - but we must trigger on 3 quality detections.

**Comment by genghisken** [comment]
OK.  On 58207 there are only two detections with pvr > 200.  We need three.  Correct me if I'm wrong.

For HKO on my test dataset (MJD 57923) the number of objects triggered are (including variable stars):
* 16,072 detections flagged (no other criteria) if the threshold is 500.
* 25,927 detections flagged (no other criteria) if the threshold is 200.

... so only a factor of 1.6 increase in the number of detections.  (The number is consistent across all association radius ranges.)

So... I think this is good enough.  I'll reset the trigger threshold now.

**Comment by smarttgit** [comment]
"_OK. On 58207 there are only two detections with pvr > 200. We need three. Correct me if I'm wrong._"

Your are correct - but there is a detection on 58205.5445666 at pvr = 267 

Therefore by 58207, the three quality detection threshold (pvr or ptr > 200) will have accumulated. 

So we would not flag on 58205 but would do on 58207 

Aslong as we flag on 58207, that is ok (for now)

**Comment by genghisken** [comment]
The trigger is not currently cumulative across multiple nights.  At the moment objects only trigger if there are three or more detections intra-night.  So, as the code currently works, it would not flag on 58207.

**Comment by smarttgit** [comment]
It's time to completely rethink. We're tweaking and not making any progress. 

Reducing pvr/ptr to less than 200 effectively just means we are saying it is no use. So let's make that decision. Get rid of a pvr and ptr cut altogether, it is not fit for purpose.  For example, this object had 4 detections at ~5-10 sigma on one night and still not getting through. 

So let's remove the pvr/ptr altogether. Looking at the plots on
https://atlas.fallingstar.com/atlas/trac/wiki/AtlasTransientServer2017

This is not going to explode the numbers. 

We need to focus on getting rid of the variable stars - this is still a bottleneck. e.g. the eyeball list contains many orphans that are really obvious stars

**Comment by katemaguire** [comment]
What login is it for that plot page? 

If the pvr/ptr cuts are removed does this mean everything (i.e. even objects with only one detection on one night) will get through or are there already some cuts in place?

**Comment by smarttgit** [comment]
not everything - we would still require minimum of 3 (>5 sigma) detections on any one night 

login - it's the internal project wiki in Hawaii. Ken can just dump in the plot here to be easier

**Comment by genghisken** [comment]
I agree - we should ditch the pvr/ptr requirement, but the difference is an order of magnitude.

This with no pvr/ptr cuts. See the N=3 curves. Use radius of 3.5 as the guide.
![image](https://user-images.githubusercontent.com/5502468/38556429-ef5381e6-3cc9-11e8-9eb1-3e69a6954326.png)

This is with pvr/ptr cuts. See the N=3 curves. Use radius of 3.5 as the guide.
![image](https://user-images.githubusercontent.com/5502468/38556532-4f0f6b22-3cca-11e8-949f-19780e985a4a.png)

**Comment by genghisken** [comment]
It's true that many of them will be variable stars, but this requires that Sherlock run on the promoted objects in order to classify them.  If we can keep up with Sherlock (running on 7-10x the numbers it's currently doing), then there's no problem with the order of magnitude change in post ingest numbers.

**Comment by smarttgit** [comment]
we have to do this - we must, must get rid of any object near a stellar source. This is the way to go

**Comment by genghisken** [comment]
OK.  I've zeroed the values for the time being.  I suspect the numbers will be manageable initially, but as more data flows the numbers will start to build.

In fact, the post ingest code actually tries to grab everything ingested in the last three days to do the cuts.  I could leave it like that to see if the numbers explode after three full nights of observing, or I could actually just cut to ingest ONLY from the last 24 hours.  If we're keeping up with the ingest, there's no reason why we couldn't just do the last 24 hours anyway.

Meanwhile the whole way we do the cutting needs to change to be done on a per-detection basis (at the point of ingest).  We're doing it in batches at the moment after ingest, but this is not compatible with LSST processing.

**Comment by thespacedoctor** [comment]
The current state of play with sherlock is that it can classify **2342 transients/minute**. So should easily cope with tens of thousands of sources. Details here:

https://github.com/thespacedoctor/sherlock/issues/50

**Comment by smarttgit** [comment]
Very good guys ! Let's give it a shot

**Comment by genghisken** [comment]
OK.  I've had to enable the 200 pvr/ptr threshold for the time being.  The rate of classifications for Sherlock is 900 transients/minute, less than half the quoted rate.  We can't keep up.  Meanwhile a whole bunch of other unexpected things went wrong - stats production fell over, and stamp rsync processes fell over for reasons I'm still trying to figure out.  Once we've caught up the last few days, I'll remove thresholds again.

**Comment by katemaguire** [comment]
Why was this ASASSN object (AT 2018arx) not triggered on? https://star.pst.qub.ac.uk/sne/atlas4/candidate/1140634800323442700/ Had 5 detections from the 3 nights ago above pvr of 200

**Comment by katemaguire** [comment]
SN 2018ast (z=0.014 galaxy) was in the garbage: https://star.pst.qub.ac.uk/sne/atlas4/candidate/1114107741244910900/ but clearly visible in the image stamps

pvr/ptr are very low, I assume because of the saturated star close by?

**Comment by genghisken** [comment]
Wasn't thrown away by an algorithm.  Is it possible that it was thrown away by an eyeballer, because there is a duplicate here in the good list already? https://star.pst.qub.ac.uk/sne/atlas4/candidate/1114108011244910100/

(Duplicates (#96) are a separate problem that requires non-trivial merge code. All duplicates in the good list will get reported to TNS - and I need to stop this happening.)

**Comment by katemaguire** [comment]
OK. The ATLAS data (or the link to the object already in the 'good' list) isn't propagated through to the ePESSTO Marshall so looked like an object that ATLAS hadn't 'discovered'.

**Comment by Lightbulb500** [comment]
The eyeballer in question was me
I mistakenly binned this one (rather than placing it in the attic) after I promoted its counterpart/duplicate ATLAS18nnj to the good list

**Comment by genghisken** [comment]
The ATLAS cuts are set such that pvr OR ptr > 50.  However, if psc > 0, the pvr and ptr triggers are reduced right down to just being non-zero.
